### PR TITLE
Introduce `bytesWritten` counter for the events endpoint

### PR DESCRIPTION
**Pull request by wavesoft**
Summary:
This commit introduces a `bytesWritten` metric for the SSE endpoint that counts the number of bytes
sent. This is to complement the PR #6286. Since this is a write-only endpoint there is no need to
introduce a `bytesRead` metric.

JIRA issues: MARATHON-8264

### CODE REVIEW

**Comment by wavesoft** [comment]
Sorryyy 😓

File: src/main/scala/mesosphere/marathon/core/event/impl/stream/HttpEventStreamServlet.scala [hunk 1]
```diff
@@ -32,6 +34,27 @@ class HttpEventSSEHandle(request: HttpServletRequest, emitter: Emitter, allowHea
...
+    * "
+    *
+    * @param eventName The name of the event, used for estimating the "name" field length
+    * @param payload The event payload, used for estimating the "data" field length
+    * @return Passes through the `payload` argument
+    */
+  private def measureFrameBytesSent(eventName: String, payload: String): String = {
+    val overhead: Long = 16 + eventName.length
+    bytesWrittenMetric.increment(payload.length + overhead)
+    payload
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/core/event/impl/stream/HttpEventStreamServlet.scala]: What is the motivation for returning the payload?

File: src/main/scala/mesosphere/marathon/core/event/impl/stream/HttpEventStreamServlet.scala [hunk 2]
```diff
@@ -42,8 +65,10 @@ class HttpEventSSEHandle(request: HttpServletRequest, emitter: Emitter, allowHea
 
   override def sendEvent(event: MarathonEvent): Unit = {
     if (subscribed(event.eventType)) {
-      if (useLightWeightEvents) blocking(emitter.event(event.eventType, event.lightJsonString))
-      else blocking(emitter.event(event.eventType, event.fullJsonString))
+      if (useLightWeightEvents) blocking(emitter.event(event.eventType,
+        measureFrameBytesSent(event.eventType, event.lightJsonString)))
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/core/event/impl/stream/HttpEventStreamServlet.scala]: I don't quite get the benefit of nesting the calls. Why not

```scala
val payload = if (useLightWeightEvents) event.lightJsonString else event.fullJsonString
measureFrameBytesSent(event.eventType, payload)
blocking(emitter.event(event.eventType, payload))
```

File: src/main/scala/mesosphere/marathon/core/event/impl/stream/HttpEventStreamServlet.scala [hunk 3]
```diff
@@ -32,6 +34,26 @@ class HttpEventSSEHandle(request: HttpServletRequest, emitter: Emitter, allowHea
...
+    * "name: <name>\n
+    *  data: <payload>\n
+    *  \n
+    * "
+    *
+    * @param eventName The name of the event, used for estimating the "name" field length
+    * @param payload The event payload, used for estimating the "data" field length
+    * @return Passes through the `payload` argument
+    */
+  private def measureFrameBytesSent(eventName: String, payload: String): String = {
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/core/event/impl/stream/HttpEventStreamServlet.scala]: Nit: I don't think the return type is correct.

Comment (wavesoft) [reply src/main/scala/mesosphere/marathon/core/event/impl/stream/HttpEventStreamServlet.scala]: yup, fixed it, forgot to push 😄

**Pull request merged**